gateway/util.go

```diff
@@ -216,3 +216,28 @@ func containsEscapedChars(str string) bool {
 
 	return str != unescaped
 }
+
+// getIpAddress returns the first non-loopback IPv4 address found. Returns error if it fails
+// to get the list of addresses. Returns empty if there's no valid IP addresses.
+// netInterfaceAddrs is used to allow mocking in the tests
+var (
+	netInterfaceAddrs = net.InterfaceAddrs
+)
+
+func getIpAddress() (string, error) {
```

Comment:
This also needs a move, and possibly ipv6 support

@jeffy-mathew internal/netutil, wdyt?

Reply:
Also, hosts may have multiple IP addresses, needs to be a form of []string as it's usually not clear which one is a "primary".

Reply:
Yes, but for the need of the ticket we only need one valid IP, we ignore loopback ones.

Reply:
> This also needs a move
indeed, I was wondering where it would fit